### ISSUE TITLE
Tests involving some floating-point numbers are unfixable

### ISSUE THREAD

**Issue by bdesham** [ianthehenry/judge#10]
I’m using Judge 2.8.1. I had a vexing test failure that I was able to boil down to Judge’s handling of floating-point numbers:

```janet
(defn one-third
  []
  (/ 1 3))

(test (one-third) 0.333333)
```

Judge wants me to make this non-change:

```
(test (one-third) 0.333333)
(test (one-third) 0.333333)
```

I’m assuming that the problem is that 1/3 is not exactly representable; the equivalent test for 1/2,

```janet
(defn one-half
  []
  (/ 1 2))

(test (one-half) 0.5)
```

passes just like you’d expect.

As a workaround, I tried to change my test to

```janet
(test (one-third) (/ 1 3))
```

but apparently Judge is too smart to let me get away with that 🙂 In my real use case, this value occurs within a reasonably complicated data structure, so it wouldn’t be easy to test the data structure excepting this one value.

**Comment by ianthehenry** [comment]
Oh yeah, sorry, that’s annoying — we should be doing an approximate equality check for floats instead of a literal `deep=`. Should be a pretty easy fix.

**Comment by ianthehenry** [comment]
Actually, maybe it would be better for Judge to serialize floats as round-trippably as possible in the expectations it generates, and to stick with exact equality. It would make some tests harder to read, but would allow you to write tests that actually depend on the precise behavior of floating points. Which would probably just be annoying most of the time? Hmm.

**Comment by bdesham** [comment]
Making the tests harder to read seems like a fair trade-off for being able to verify the exact behavior of the program. But either solution seems reasonable to me!

**Comment by ianthehenry** [comment]
Okay! I believe that I fixed this in a nice way in version 2.8.2. Let me know if it gives you any more guff.

**Comment by bdesham** [comment]
Yes, it works like a charm now. Thank you very much! A+ customer service, would depend again.
